storage/storage_image.go

```diff
@@ -137,10 +137,9 @@ func (s *storageImageDestination) putBlob(stream io.Reader, blobinfo types.BlobI
 		Digest: "",
 		Size:   -1,
 	}
-	// Try to read an initial snippet of the blob.
-	header := make([]byte, 10240)
-	n, err := stream.Read(header)
-	if err != nil && err != io.EOF {
+	buf := [archive.HeaderSize]byte{}
+	n, err := io.ReadAtLeast(stream, buf[:], len(buf))
```

Comment:
Is there a difference between `buf[:]` and `buf` here? Both are slices, and `stream.Read(slice)` was fine above.